lib/elixir/test/elixir/kernel/errors_test.exs

```diff
@@ -648,6 +648,14 @@ defmodule Kernel.ErrorsTest do
     assert_compile_fail CompileError,
       "nofile:1: cannot use ^x outside of match clauses",
       'x = 8; <<a, b::size(^x)>> = <<?a, ?b>>'
+
+    assert_compile_fail CompileError,
+      "nofile:2: size in bitstring expects an integer or a variable as argument, got: bar()",
+      '''
+      defmodule Kernel.ErrorsTest.InvalidSizeInBitStrings do
+        def foo(<<_::size(bar)>>), do: :ok
```

Comment:
You're right, I was following the exact error described in the issue, but it doesn't matter here. Fixed on dc0e2df.